### PR TITLE
Add logic to setup and execute CCR perf test suite

### PR DESCRIPTION
**Pull request by ankitkala**
Signed-off-by: Ankit Kala <ankikala@amazon.com>

### Description
This changes adds support to run the CCR Performance suite. 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: perf_workflow/run_perf_suite/ccr_perf_test.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        root.addHandler(handler)
+
+    def workload_options(self, leaderCluster, followerCluster):
+        telemetry_options = {
+            "ccr-stats-sample-interval": 1,
+            "ccr-stats-indices": {"follower": [self.test_config["Workload"]]},
+        }
+        workload_options = {
+            "telemetry-params": json.dumps(telemetry_options),
+            "workload-params": "number_of_shards:{},number_of_replicas:{}".format(
```
Comment (saikaranam-amazon) [inline perf_workflow/run_perf_suite/ccr_perf_test.py:198]: Could we pass codec index setting as well to force the default (optimized for speed)?

File: perf_workflow/run_perf_suite/ccr_perf_test.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                        )
+                        self.setup_replication(
+                            leaderCluster, followerCluster, self.test_config["Workload"]
+                        )
+                        # Setup args for Perf Suite.
+                        target_hosts = {
+                            "default": [leaderCluster.endpoint_with_port()],
+                            "follower": [followerCluster.endpoint_with_port()],
+                        }
+                        workload_options = self.workload_options(
```
Comment (saikaranam-amazon) [inline perf_workflow/run_perf_suite/ccr_perf_test.py:106]: As discussed - lets makes sure, we have the same instance type across the runs and the workload params are configured to match the instance type.

File: perf_workflow/run_perf_suite/ccr_perf_test.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                ), "Replication lag p100 was greater than a minute"
+                self.test_succeeded = True
+                logging.info(f"Test succeeded for {self.description}")
+
+    def verify_doc_count(self, leaderCluster, followerCluster):
+        index_name = self.test_config["Workload"]
+        path = f"/{index_name}/_count?format=json"
+
+        leader_url = "".join([leaderCluster.endpoint_with_port(), path])
+        leader_resp = requests.get(url=leader_url, auth=self.CREDS, verify=False)
```
Comment (soosinha) [inline perf_workflow/run_perf_suite/ccr_perf_test.py:164]: Where is `CREDS` defined?

File: perf_workflow/run_perf_suite/ccr_perf_test.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+                "Unable to update trigger autofollow replication: {}".format(
+                    resp.reason
+                )
+            )
+
+    def setup_seed_nodes(self, leaderCluster, followerCluster):
+        requests.packages.urllib3.disable_warnings()
+        # Fetch seed nodes from leader cluster
+        url = "".join([leaderCluster.endpoint_with_port(), "/_cat/nodes?format=json"])
+        resp = requests.get(url=url, auth=HTTPBasicAuth("admin", "admin"), verify=False)
```
Comment (soosinha) [inline perf_workflow/run_perf_suite/ccr_perf_test.py:259]: Can you declare username/password and then use?

File: run_perf_test.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (ankitkala) [reply run_perf_test.sh:1]: Sure.

File: perf_workflow/run_perf_suite/ccr_perf_test.py (hunk 1 above)
Comment (ankitkala) [reply perf_workflow/run_perf_suite/ccr_perf_test.py:198]: Tried adding this in workload params but the OSB test run failed. Will try to get that fixed and do this change in a follow up PR.

File: perf_workflow/run_perf_suite/ccr_perf_test.py (hunk 2 above)
Comment (ankitkala) [reply perf_workflow/run_perf_suite/ccr_perf_test.py:106]: Yep. All tests are being run on `m5.2xlarge`

File: perf_workflow/run_perf_suite/ccr_perf_test.py (hunk 3 above)
Comment (ankitkala) [reply perf_workflow/run_perf_suite/ccr_perf_test.py:164]: My bad. Got removed accidentally. Added it back now.

File: perf_workflow/run_perf_suite/ccr_perf_test.py (hunk 4 above)
Comment (ankitkala) [reply perf_workflow/run_perf_suite/ccr_perf_test.py:259]: Replaced with `self.CREDS`

**Review by soosinha — APPROVED** [review]

**Review by saikaranam-amazon — APPROVED** [review]

**Review by gbbafna — APPROVED** [review]
We will move to another repo as a quick follow up. Approving for now.

**Pull request merged**